docs/sending-data-to-launchable/running-under-restricted-networks.md

```diff
@@ -0,0 +1,88 @@
+# Running under restricted networks
+
+{% hint style="info" %}
+This page relates to [#recording-builds](./#recording-builds "mention").
+{% endhint %}
+
+Launchable CLI requires the Internet connection, but your environment might have
+a limited Internet reachability. This page explains how to deal with it in
+common cases.
+
+## Outbound traffic control
+
+Networks can have a network policy that limits the connections to certain IP
+addresses and ports. In this case, you need to allowlist Launchable's API server
+IP addresses. We have an API endpoint that has a stable IP address.
+
+* Domain: `api-static.mercury.launchableinc.com`
+* IP Addresses: `13.248.185.38` and `76.223.54.162`
+* Port: `443` (HTTPS)
+
+Once your network is configured to allow traffic to the IP addresses above, you
+can configure the Launchable CLI to use that endpoint:
+
+```bash
+export LAUNCHABLE_BASE_URL="https://api-static.mercury.launchableinc.com"
+launchable verify
+```
+
+## Git repository in an Internet unreachable environment
+
+Your repository might have no Internet access for security reasons. For example,
+the CI workflow would look like:
+
+1. Check out the Git repository in a machine with no Internet connection.
+2. Build the artifacts in the machine.
+3. Transfer the artifacts to a test environment that has the Internet
+   connection.
+4. Run tests against the artifacts in the test environment.
+
+In this case, you can save the `git log` output to a file and transfer that file
+to the test environment so that Launchable CLI can see the changes under test.
+
+In the build machine, you will need to run a `git log` command with
+`--pretty='format:{"commit": "%H", "parents": "%P", "authorEmail": "%ae",
+"authorTime": "%aI", "committerEmail": "%ce", "committerTime": "%cI"}'
+--numstat`.
+
+```bash
+# Check out the repository and build artifacts
+git clone https://git.internal.example.com/repo
+cd repo
+make all
+
+# Save git log data to a file. Using --max-count to limit the number of commits.
```

Comment:
Since it's important, we should include `--max-count` in the CLI reference and on line 43 in this file.

I'd like to provide guidance on the right value to choose, but that seems impossible.

Because of this, perhaps we could recommend using `--since` instead of `--max-count`? e.g. --since="2 days ago"

Reply:
I think the time range is also tricky (consider that the test run after a long weekend). I just gave 10-20 as a ballpark figure. This should be suffice in many cases.

Reply:
OK! We can always adjust later.